### PR TITLE
E2E tests: Onboarding Step 3 - Confirm store requirements

### PR DESCRIPTION
**Pull request by ianlin**
### Changes proposed in this Pull Request:

Closes part of #2070, this PR adds E2E tests for [📌 Onboarding Step 3 - Confirm store requirements][1].

### Detailed test instructions:

1. Follow these steps to set up the e2e env: https://github.com/woocommerce/google-listings-and-ads#e2e-testing
2. Run `npm run test:e2e -- ./tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js`, the test should pass.
3. Or, run `npm run test:e2e-dev -- ./tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js`, check the test in the chromium step by step.
4. Run `npm run test:e2e` to make sure all tests are passed.

### Changelog entry

> Dev - E2E - Onboarding Step 3 - Confirm store requirements

[1]: https://github.com/woocommerce/google-listings-and-ads/issues/2070#onboarding-step-3

### CODE REVIEW

File: tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+ */
+import { LOAD_STATE } from '../../utils/constants';
+import StoreRequirements from '../../utils/pages/setup-mc/step-3-confirm-store-requirements';
+
+test.use( { storageState: process.env.ADMINSTATE } );
+
+test.describe.configure( { mode: 'serial' } );
+
+/**
+ * @type {import('../../utils/pages/setup-mc/step-1-set-up-accounts.js').default} storeRequirements
```
Comment (jorgemd24) [inline tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js]: This type is incorrect, should be: `step-3-confirm-store-requirements.js`

File: tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+	/**
+	 * Go to the set up mc page.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async goto() {
+		await this.page.goto(
+			'/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc&google-mc=connected',
+			{ waitUntil: LOAD_STATE.NETWORK_IDLE }
```
Comment (jorgemd24) [inline tests/e2e/utils/pages/setup-mc/step-3-confirm-store-requirements.js]: As I https://github.com/woocommerce/google-listings-and-ads/pull/2088#discussion_r1324325817 in the other PR, not sure if we should use NETWORK_IDLE.

File: tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+					const checkboxes =
+						storeRequirements.getPrelaunchChecklistCheckboxes();
+
+					for ( let i = 2; i <= 3; i++ ) {
+						const checkbox = await checkboxes.nth( i );
+						await checkbox.click();
+						await expect( checkbox ).toBeDisabled();
+						await expect( checkbox ).toBeChecked();
+					}
```
Comment (jorgemd24) [inline tests/e2e/specs/setup-mc/step-3-confirm-store-requirements.test.js:412]: What do you think if we test whether the following requests are being made: `POST wc/gla/mc/settings` and, when we click "Continue," we also send the request `POST wc/gla/mc/contact-information`? Should we verify this one as well?

**Review by jorgemd24 — COMMENTED** [review]
Thanks, @ianlin! The PR covers most of the scenarios, but I'm uncertain whether we should also check if certain requests are being triggered, I left a comment about it.

**Comment by ianlin** [comment]
Hey @jorgemd24, thanks for reviewing and the nice suggestions. I've updated the code based on your suggestions and it's ready for another round of review.

**Review by jorgemd24 — APPROVED** [review]
Thanks @ianlin for the changes! LGTM!

**Pull request merged**